server/py/services/api/crud/projects.py

```diff
@@ -347,11 +347,18 @@ async def list_allowed_project_names_with_creation_time(
             format_=mlrun.common.formatters.ProjectFormat.name_and_creation_time,
             **project_filters,
         )
-        return await framework.utils.auth.verifier.AuthVerifier().filter_projects_by_permissions(
+        allowed_project_names = await framework.utils.auth.verifier.AuthVerifier().filter_projects_by_permissions(
             [project[0] for project in projects_output.projects],
             auth_info,
             action=action,
         )
+        # Filter the original list based on allowed names
+        # we need to return full project object (not only project names)
```

Comment:
You don't need the full project object (i.e. `Project`), you need the tuple that includes the name + creation time.
(As the function name suggests, along with its return type`list[tuple[str, datetime.datetime]]`)

Reply:
@TomerShor full project object means what we have in `projects_output.projects`